### ISSUE TITLE
  nbsp.identify_data_from_index_search(fingerprint_hash)>>>>> RuntimeError: Unknown format

### ISSUE THREAD

**Issue by GeneralTITO** [Andoreatta/nbsp-python#3]
uint32_t NBioBSP_IdentifyDataFromIndexSearchDB(std::string firText) {
    NBioAPI_FIR_TEXTENCODE mouldFirText;
    mouldFirText.TextFIR = &firText[0];

    NBioAPI_INPUT_FIR inputFIR;
    inputFIR.Form = NBioAPI_FIR_FORM_TEXTENCODE;
    inputFIR.InputFIR.TextFIR = &mouldFirText;

    NBioAPI_INDEXSEARCH_FP_INFO fpInfo;

    NBioAPI_RETURN nbioApiReturn = NBioAPI_IdentifyDataFromIndexSearchDB(nbioApiHandle, &inputFIR, 5, &fpInfo, NULL);
    check_nbioapi_error(nbioApiReturn);

    return fpInfo.ID;
}
I don't know c++, but looking over it, maybe it's on line 3 of what I sent here, having to remove "[0]".

**Comment by Andoreatta** [comment]
I'll take a look as soon as I can.

It has [0] because essentially, C++ strings are an array of chars, `&firText[0]` is used to get the pointer to the first char of the array, which basically is getting a pointer to the string/array itself. `NBioAPI_FIR_TEXTENCODE` struct expects a pointer to the char array (`char*`) for the TextFIR attribute.

I did this because the library for nbiobsp is C and I had to find ways to interface with it.
 
I will be taking a look and will simplify it with `c_str()`, as I just remembered this exists, but anyways, it should be working normally, did you test the tkinter example?

**Comment by Andoreatta** [comment]
And, the error it returned is "unknown format" which means the lib thinks theres something wrong with the hash you passed.

**Comment by Andoreatta** [comment]
OBS: se quiser falar em pt-br

**Comment by GeneralTITO** [comment]
Sim, testei no exemplo tkinter também. Funciona tudo, menos essa parte de verificar no indexsearch, que também aparece em baixo dizendo que o formato é desconhecido.

**Comment by Andoreatta** [comment]
O verify retorna um erro ou fica na tela de captura esperando? O que pode acontecer é um timeout.
Você baixou do pypi ou compilou por si mesmo? Pode me mandar um printscreen do exemplo do tkinter dando errado? Ou até mesmo do seu código?

**Comment by GeneralTITO** [comment]
Sim, gravei até um vídeo para demonstrar todo o processo. https://www.youtube.com/watch?v=u_Cn4mISvfs

**Comment by GeneralTITO** [comment]
Achei esse repo aqui, mas não consegui instalar na minha maquina: https://github.com/xxpauloxx/python-nbiobsp.
Talvez ajude. E tem uma função 'match' que seria uma boa.

**Comment by Andoreatta** [comment]
Certo, irei investigar, retorno ainda hoje ou no máximo amanhã.
A função match, nesse caso nomeado verify, está ali no tkinter, você mesmo executou e retornou "valid fingerprint"

**Comment by GeneralTITO** [comment]
A verify apenas verifica a ultima do handle. A de 'match' seria para verificar se duas hashs seriam do mesmo dedo, podendo pegar uma hash de um banco de dados por exemplo.

**Comment by Andoreatta** [comment]
Certo, fiz testes puxando a lib com o `pip install nbsp-python` em 3 computadores e não obtive erro.
E a questão estranha é que a lógica do código, com aquela sintaxe, está correta.
1. Por mais irrelevante, peço que você execute o `pip install nbsp-python` sem o underline como você fez no vídeo. Apenas pra termos certeza.
2. Teste em um computador alternativo, apenas pra termos certeza, foi o que eu fiz aqui.
3. Se você compilou diretamente do github, você utilizou o GCC ou MSVC?

**Comment by GeneralTITO** [comment]
1- continua dando o mesmo erro
2- só tenho esse
3- acho q o msvc. Não sei muito sobre compiladores, mas baixei ferramentas de build e compilação do msvc pelos componentes individuais do visual studio.

Estava mexendo no código e copiei a função match do outro repo para a nbsp-python antes de compilar e acabou resolvendo o problema de comparar as hashs por aqui. 
Mas se quiser continuar investigando o erro, qualquer pergunta eu respondo.

**Comment by Andoreatta** [comment]
Definitivamente irei investigar, mas devido as demandas de agora na empresa, estou deixando um pouco de lado.
Talvez eu refatore, mas tenho que me preocupar em não quebrar aplicações feitas com a lib também. Agradeço seu input sobre o caso.

Percebi que funciona no windows 10, mas no 11 tem problemas em algumas das vezes, não faço ideia do motivo.

Vou deixar o issue em aberto até conseguirmos resolver o problema.
